lib/SPIRVOp.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2019 The Clspv Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/IR/Attributes.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Type.h"
+#include "llvm/IR/Value.h"
+
+#include "spirv/1.0/spirv.hpp"
+
+namespace clspv {
+
+using namespace llvm;
+
+// Insert a call to a specific SPIR-V instruction after Insert
```

Comment:
Please make this comment a bit more verbose. Mention unique name generation and that `Attributes` must be function attributes.

Reply:
I should indeed have done a better job there. I've just pushed a commit with the requested changes. I've also added a warning about the symbol table being changed as that tripped me on the very first use.